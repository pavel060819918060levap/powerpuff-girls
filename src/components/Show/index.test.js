import React from 'react';
import { shallow } from 'enzyme';
import { Show } from './';

it('should not render show when it is not passed', () => {
  const mockOnInitCallback = jest.fn();
  const component = shallow(<Show
    onInit={mockOnInitCallback}
    match={{
      params: {
        showId: '20',
      },
    }}
  />);

  expect(mockOnInitCallback).toHaveBeenCalledWith('20');
  expect(component).toMatchSnapshot();
});

it('should render show when it is passed', () => {
  const mockOnInitCallback = jest.fn();
  const component = shallow(<Show
    onInit={mockOnInitCallback}
    match={{
      params: {
        showId: '20',
      },
    }}
    details={{
      show: {
        name: 'Test show',
        image: {
          medium: 'http://testurl.com',
        },
        summary: '<p>This is test show</p>',
        id: 123,
      },
      episodes: [
        {
          id: 30,
          season: 2,
          number: 3,
          image: {
            medium: 'http://testurl.com',
          },
        },
        {
          id: 31,
          season: 1,
          number: 4,
          image: {
            medium: 'http://testurl.com',
          },
        },
      ],
    }}
  />);

  expect(mockOnInitCallback).toHaveBeenCalledWith('20');
  expect(component).toMatchSnapshot();
});

it('should not render show when there is some error', () => {
  const mockOnInitCallback = jest.fn();
  const component = shallow(<Show
    onInit={mockOnInitCallback}
    match={{
      params: {
        showId: '20',
      },
    }}
    error={true}
    details={{
      show: {
        name: 'Test show',
        image: {
          medium: 'http://testurl.com',
        },
        summary: '<p>This is test show</p>',
        id: 123,
      },
      episodes: [
        {
          id: 30,
          season: 2,
          number: 3,
          image: {
            medium: 'http://testurl.com',
          },
        },
        {
          id: 31,
          season: 1,
          number: 4,
          image: {
            medium: 'http://testurl.com',
          },
        },
      ],
    }}
  />);

  expect(mockOnInitCallback).toHaveBeenCalledWith('20');
  expect(component).toMatchSnapshot();
});
