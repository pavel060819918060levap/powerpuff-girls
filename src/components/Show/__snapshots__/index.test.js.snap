// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should not render show when it is not passed 1`] = `<styled.div />`;

exports[`should not render show when there is some error 1`] = `
<styled.div>
  <Error
    title="Some error"
  >
    Something went wrong... :(
  </Error>
</styled.div>
`;

exports[`should render show when it is passed 1`] = `
<styled.div>
  <SectionTitle>
    Test show
  </SectionTitle>
  <styled.div>
    <styled.img
      src="http://testurl.com"
    />
    <styled.div
      dangerouslySetInnerHTML={
        Object {
          "__html": "<p>This is test show</p>",
        }
      }
    />
  </styled.div>
  <SectionSubtitle>
    Episodes
  </SectionSubtitle>
  <styled.ul>
    <styled.li
      key="30"
    >
      <Link
        replace={false}
        to="/show/123/2/3"
      >
        <styled.img
          src="http://testurl.com"
        />
      </Link>
    </styled.li>
    <styled.li
      key="31"
    >
      <Link
        replace={false}
        to="/show/123/1/4"
      >
        <styled.img
          src="http://testurl.com"
        />
      </Link>
    </styled.li>
  </styled.ul>
</styled.div>
`;
